frontend/catalyst/jax_primitives.py

```diff
@@ -203,6 +204,39 @@ def _obs_lowering(aval):
 vjp_p.multiple_results = True
 adjoint_p = jax.core.Primitive("adjoint")
 adjoint_p.multiple_results = True
+print_p = jax.core.Primitive("debug_print")
+print_p.multiple_results = True
+
+
+#
+# print
+#
+def debug_print(x):
+    """A runtime print function for Catalyst."""
+
+    if isinstance(x, jax.core.Tracer):
+        prim = print_p.bind(x)
+    else:
+        prim = print_p.bind(string=str(x))
+
+    return prim
```

Comment:
thought: @dime10 a question: how easy would it be (in a future story/PR) to support something like f-strings or format strings? E.g.,

```python
catalyst.debug_print("Step: {i}  Loss: {x}", i=i, x=cost_val)
```

This way we can support something _close to_ f-strings, provide more flexibility to the user. This is a pattern I often see/use in optimization, e.g.:

<img width="718" alt="image" src="https://github.com/PennyLaneAI/catalyst/assets/2959003/d7081165-5eb0-4fee-b6d8-913cd05f0737">

Reply:
Not sure, I've never looked at an implementation of `format` 😅
I suppose it could be handled in Python, and broken up into multiple Catalyst print statements with a mixture of fixed string arguments and dynamic array arguments. Or it could be fully implemented in the runtime, with a variable argument signature (like `printf`) 🤔

Reply:
I like the former is closer to the Python approach:
- Evaluate expressions inside `{}` and convert them to strings
- Apply formatting rules and concatenate the results
- Print the result with the `print()` function

Reply:
Added to the backlog :)